src/ripple/ledger/CashDiff.h

```diff
@@ -62,7 +62,7 @@ class CashDiff
 public:
     CashDiff() = delete;
     CashDiff (CashDiff const&) = delete;
-    CashDiff (CashDiff&& other);
+    CashDiff (CashDiff&& other) noexcept;
```

Comment:
We could `noexcept = default` this one while we're here.

Reply:
Right. I went ahead and default the dtor too while in the neighborhood.

Reply:
@HowardHinnant I don't think we actually want to `noexcept = default` it here. 

My reasoning: to default it would require exposing the complete type if `Impl` when parsing the `CashDiff` type. This is because the default move ctor would require knowing the full type of the `std::unique_ptr<Impl>`.  Right now we forward declare `Impl` so we have an incomplete type. I suspect this is also the reason why the destructor (though trivial) is defined out of line -- to avoid having the complete type.

Thoughts?

Reply:
Good thing you're on the ball! Sounds completely right. 👍